File: Application/Sources/Helpers/Categories/UIImage+PlaySRG.m
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

#import "UIImage+PlaySRG.h"

NSString *FilePathForImagePlaceholder(ImagePlaceholder imagePlaceholder)
{
    switch (imagePlaceholder) {
        case ImagePlaceholderMedia: {
            return [NSBundle.mainBundle pathForResource:@"placeholder_media" ofType:@"pdf"];
            break;
        }
            
        case ImagePlaceholderMediaList: {
            return [NSBundle.mainBundle pathForResource:@"placeholder_media_list" ofType:@"pdf"];
            break;
        }
            
        case ImagePlaceholderNotification: {
            return [NSBundle.mainBundle pathForResource:@"placeholder_notification" ofType:@"pdf"];
            break;
        }
            
        default: {
            return nil;
            break;
        }
    }
}

UIImage *YouthProtectionImageForColor(SRGYouthProtectionColor youthProtectionColor)
{
    switch (youthProtectionColor) {
        case SRGYouthProtectionColorYellow: {
            return [UIImage imageNamed:@"youth_protection_yellow"];
            break;
        }
            
        case SRGYouthProtectionColorRed: {
            return [UIImage imageNamed:@"youth_protection_red"];
            break;
        }
            
        default: {
            return nil;
            break;
        }
    }
}

UIImage *ImageForBlockingReason(SRGBlockingReason blockingReason)
{
    switch (blockingReason) {
        case SRGBlockingReasonGeoblocking: {
            return [UIImage imageNamed:@"geoblocked"];
            break;
        }
            
        case SRGBlockingReasonLegal: {
            return [UIImage imageNamed:@"legal"];
            break;
        }
            
        case SRGBlockingReasonAgeRating12:
        case SRGBlockingReasonAgeRating18: {
            return [UIImage imageNamed:@"age_rating"];
            break;
        }
            
        case SRGBlockingReasonStartDate:
        case SRGBlockingReasonEndDate:
        case SRGBlockingReasonNone: {
            return nil;
            break;
        }
            
        default: {
            return [UIImage imageNamed:@"generic_blocked"];
            break;
        }
    }
}

File: Application/Sources/Helpers/Extensions/UIImage+PlaySRG.swift
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

import SRGDataProviderModel
import UIKit

extension UIImage {
    /**
     *  Return youth protection image associated with a color, if any.
     */
    @objc static func image(for youthProtectionColor: SRGYouthProtectionColor) -> UIImage? {
        switch youthProtectionColor {
        case .yellow:
            UIImage(resource: .youthProtectionYellow)
        case .red:
            UIImage(resource: .youthProtectionRed)
        default:
            nil
        }
    }

    /**
     *  Return the standard image to be used for a given blocking reason, if any.
     */
    static func image(for blockingReason: SRGBlockingReason) -> UIImage? {
        switch blockingReason {
        case .geoblocking:
            UIImage(resource: .geoblocked)
        case .legal:
            UIImage(resource: .legal)
        case .ageRating12, .ageRating18:
            UIImage(resource: .ageRating)
        case .startDate, .endDate, .none:
            nil
        default:
            UIImage(resource: .genericBlocked)
        }
    }
}
